src/CoreSystems/Application.cc

```diff
@@ -35,6 +38,25 @@ namespace Espert
 
     void Application::events_manager(Event &e)
     {
-        Event::try_hanlder<WindowClosedEvent>(e, BIND_EVENT_FOR_FUN(on_window_closed));
+        Event::try_hanlder<WindowClosedEvent>(e, ESP_BIND_EVENT_FOR_FUN(Application::on_window_closed));
+
+        for (auto it = _m_layer_stack.end(); it != _m_layer_stack.begin(); )
```

Comment:
Please use references for iterator types in range based for loop.
Also since C++20 iterating in reverse order in range based for loops can be done more simply:

for (auto const& x : range | std::views::reverse)
{
    foo(x);
}

Reply:
Co do referencji to się zgadzam ale ja bym w sumie został przy takim tradycyjnym przejściu iteratorem bo przynajmniej wiadomo o co chodzi haha

Reply:
ENGLISH PLEASE!!!!